### PR TITLE
Adding the option to BYO ElasticSearch

### PR DESCRIPTION
**Pull request by markmark206**
Adding the option to Bring Your Own ElasticSearch (`helm install -f values/values.elasticsearch.yaml temporaltest .`)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,17 +24,30 @@ Download Helm dependencies:
...
 To install Temporal with all of its dependencies, including Cassandra and ElasticSearch, run this command:
 
 ```bash
 ~/temporal-helm$ helm install temporaltest . --timeout 900s
 ```
 
-#### MySQL Running Separately
+#### Bring Your Own ElasticSearch
+
+You might already be operating an intance of ElasticSearch that you want to use with Temporal.
```
Comment (alexshtin) [inline README.md:39]:
```suggestion
You might already be operating an instance of ElasticSearch that you want to use with Temporal.
```

File: README.md [hunk 2]
```diff
@@ -24,17 +24,30 @@ Download Helm dependencies:
...
 ```bash
 ~/temporal-helm$ helm install temporaltest . --timeout 900s
 ```
 
-#### MySQL Running Separately
+#### Bring Your Own ElasticSearch
+
+You might already be operating an instance of ElasticSearch that you want to use with Temporal.
+
+To do so, fill in the relevant configuration values in `values/values.cassandra.yaml`, and pass the file to 'helm install'.
```
Comment (kfoxb) [inline README.md:41]: I think this should say `values/values.elasticsearch.yaml` right?

File: README.md [hunk 3]
```diff
@@ -24,17 +24,30 @@ Download Helm dependencies:
...
+
+Exmaple:
```
Comment (kfoxb) [inline README.md:43]: typo here

**Comment by kfoxb** [comment]
I think you may also want to add a condition to the Chart.yaml under the elasticsearch entry just like Cassandra currently has :-)

**Comment by kfoxb** [comment]
> I think you may also want to add a condition to the Chart.yaml under the elasticsearch entry just like Cassandra currently has :-)

Ignore this, sorry, I didn't see you've already set this. My branch was a little out of date.

**Comment by kfoxb** [comment]
I think it'd be nice to update to version in Chart.yaml file with this change. :-)

File: templates/server-deployment.yaml [hunk 4]
```diff
@@ -70,11 +70,11 @@ spec:
                 fieldRef:
                   fieldPath: status.podIP
             - name: ENABLE_ES
-              value: "{{ $.Values.elasticsearch.enabled }}"
+              value: "{{ or $.Values.elasticsearch.enabled $.Values.elasticsearch.hosted }}"
             - name: ES_SEEDS
-              value: elasticsearch-master-headless
+              value: {{ $.Values.elasticsearch.host }}
```
Comment (kfoxb) [inline templates/server-deployment.yaml:75]: I think this breaks the default setup, since a host and port isn't provided in the base values.yaml :-)

Comment (markmark206) [reply templates/server-deployment.yaml:75]: thank you, fixed!

File: README.md (hunk 3 above)
Comment (markmark206) [reply README.md]: thank you! fixed creative spelling here. ;)

File: README.md (hunk 2 above)
Comment (markmark206) [reply README.md]: duh. ; ) ) thank you, fixed!

**Comment by markmark206** [comment]
thank you @kfoxb for the review!

**Comment by markmark206** [comment]
addressed pr feedback and added a test pipeline for this configuration.

**Pull request merged**